process_request/helpers.py

```diff
@@ -3,6 +3,164 @@
+
+
+def get_instance_data(instance, type):
+    """Takes ArchivesSpace instance information and returns a dictionary of selected instance information.
+
+    Args:
+        instance (dict): ArchivesSpace instance information that has resolved top containers and digital objects.
+        type (string): a string representation of an instance type.
+
+    Returns:
+        instance_data (dict): returns a dict container instance type, indicator, location, and barcode for the instance.
+    """
+    instance_data = {}
+    instance_data["instance_type"] = type
+    instance_data["indicator"] = get_container_indicators(instance)
+    if type == "digital_object":
+        digital_object = instance.get("digital_object").get("_resolved")
+        instance_data["location"] = get_file_versions(digital_object)
+        instance_data["barcode"] = None
+    else:
+        top_container = instance.get("sub_container").get("top_container").get("_resolved")
+        instance_data["location"] = get_location(top_container)
+        instance_data["barcode"] = top_container.get("barcode")
+    return instance_data
+
+
+def get_preferred_format(instances):
+    """Iterates over instances and gets the preferred delivery format based on delivery formats set in the settings.
+
+    Args:
+        instances (list): a list of ArchivesSpace instance dictionaries.
+
+    Returns:
+        preferred (list): a list of string instance information retrieved by get_instance_data.
+    """
+    instance_types = []
+    [instance_types.append((i["instance_type"])) for i in instances]
+    if "digital_object" in instance_types:
+        preferred = [get_instance_data(i, i["instance_type"]) for i in instances if i["instance_type"] == "digital_object"]
+    elif "microform" in instance_types:
+        preferred = [get_instance_data(i, i["instance_type"]) for i in instances if i["instance_type"] == "microform"]
+    elif "microfilm" in instance_types:
+        preferred = [get_instance_data(i, i["instance_type"]) for i in instances if i["instance_type"] == "microfilm"]
+    elif "mixed materials" in instance_types:
+        preferred = [get_instance_data(i, i["instance_type"]) for i in instances if i["instance_type"] == "mixed materials"]
+    else:
+        return None
+    return preferred
+
+
+def set_preferred_data(data, indicator=None, type=None, location=None):
+    """Takes a dictionary and sets arguments to key, value pairs in that dictionary.
+
+    Args:
+        data (dict): the dictionary to write to.
+        indicator (string): string representation of a container type and indicator
+        type (string): string representation of an instance type
+        location (string): string representation of a container's location
+
+    Return:
+        data (dict): returns an updated dictionary with new key, value pairs
+    """
+    data['preferred_container'] = indicator
+    data['preferred_format'] = type
+    data['preferred_location'] = location
+    return data
```

Comment:
I'm not sure I understand the utility of this, but that also may be tied to my confusion about what's happening in the main `get_data` method.

Reply:
See comment below. I originally didn't have this as a separate method, and I'm happy to remove it if we don't care about some of those key/value pairs showing up in the returned data.